user-guide/consumers.md

```diff
@@ -200,14 +200,34 @@ Consumer.start('account:command')
 
 In the above example, the consumer's position stream would be `account:command-someConsumer`.
 
+Consumers can also be assigned an identifier when they are started. If an identifier macro is also declared on the consumer class, the one given when starting the consumer is chosen.
+
+``` ruby
+Consumer.start('account:command', identifier: 'otherIdentifier')
+```
+
+## Conditions
+
+Since the consumer reads the given stream using a SQL query, that query can be extended by the `condition` keyword argument. This further constrains the messages read by the consumer beyond selecting only the messages of the stream being consumed. For instance, this allows a consumer to only read messages that correlate to a given category, or only a subset of the streams within a category for parallel processing across multiple consumers.
+
+```ruby
+SomeConsumer.start('someCategory', condition: "metadata->>'correlationStreamName' like 'otherCategory%'");
+```
+
+In the above example, the consumer will only read messages from `someCategory` that correlate to `otherCategory`.
+
+::: danger
+Usage of this feature should be treated with caution, as it can cause messages to be read by consumers in a different order in which they were written. Idempotence techniques which rely on processing messages in-order therefore may not work in conjuction with `conditions`. Therefore, it is recommended that this feature be used with care, and only in cases where the ramifications are fully understood.
+:::
+
 ## Constructing Consumers
 
 In general, it's not necessary to construct a consumer. The general use case of a consumer is to invoke its `start` method.
 
 A consumer can be constructed with its `build` method.
 
 ``` ruby
-self.build(stream_name, poll_interval_milliseconds: 100, batch_size: 1000, position_update_interval: 100, position_store: nil, session: nil)
+self.build(stream_name, poll_interval_milliseconds: 100, batch_size: 1000, position_update_interval: 100, settings: nil, condition: nil)
```

Comment:
Again, I think settings should be the last argument in the documentation (if not the signature definition)